### PR TITLE
[WIP] Update RichHelpFormatter to buffer and render later

### PR DESCRIPTION
**Pull request by dwreeves**
This PR closer aligns the `RichHelpFormatter` with how the base Click `HelpFormatter` works in regards to eager printing. This is in relation to issue #48 and a request from @ofek.

### CODE REVIEW

File: src/rich_click/rich_command.py [hunk 1]
```diff
@@ -180,15 +182,9 @@ def main(
...
-
-                if self.context_class.errors_in_output_format:
-                    export_console_as = self.context_class.export_console_as
-                else:
-                    export_console_as = None
-
-                rich_format_error(e, formatter, export_console_as=export_console_as)
+                formatter = self._error_formatter()
+                formatter.write_error(e)
+                click.echo(formatter.getvalue(), file=sys.stderr)
```
Comment (ofek) [inline src/rich_click/rich_command.py:187]: Out of curiosity, what do you think about never using Click to output and relying exclusively on Rich? That's actually what I assumed was happening already. Click uses `colorama` on Windows which is basically unmaintained and has some odd side effects. I would love for Click to stop using that eventually and actually remove terminal capabilities by default (see this [closed issue](https://github.com/pallets/click/issues/2776)).

Comment (dwreeves) [reply src/rich_click/rich_command.py:187]: That's a good suggestion and I'll consider it. Rich does have its own legacy Windows color handling, so wrapping in `click.echo()` should not be necessary for that situation. I need to make sure the other small benefits of `echo()` are also being handled by Rich in how it renders the string generated by `console.print()`.

Comment (dwreeves) [reply src/rich_click/rich_command.py]: So, I don't want to do this personally. I'm open to PRs if anyone else wants it. #233

Comment (dwreeves) [reply src/rich_click/rich_command.py]: Basically, there are a handful of minor things I ran into that suggest to me this could be a very minorly annoying breaking change, and it's a lot of work to "fully" implement for something that doesn't impact the majority of both developers or even users. If there are fundamental issues with Click's use of Colorama, I would probably more likely bring it up with Click themselves. None of that is to say the feature wouldn't be welcome, of course, but I will not personally be prioritizing it over other things I am more interested in prioritizing.

Comment (ofek) [reply src/rich_click/rich_command.py]: Did you discover any missing functionality or edge case handling on Rich's side?

Comment (dwreeves) [reply src/rich_click/rich_command.py]: No, I discovered it changes my unit-tests from being colorless to having colors with the config we currently have. That is enough for me to believe the change can be classified as "minorly annoying breaking" (imagine a person's unit tests break the same ways ours did, in a cryptic way that they don't know how to make work in the old way, because of a change from click.echo to rich.print in a minor release; they would probably be rightly annoyed.) This would make it inappropriate for anything other than a major release if we make `rich.print()` the default, rather than as a config option that's turned off by default (which would be in-scope for a minor release).

**Comment by ofek** [comment]
Do you know when this might be released?

**Comment by dwreeves** [comment]
Oh neat, looks like a cool project. And yeah this is definitely a huge problem with `stdio` transit. I think the same general idea of:

```python
    try:
        import rich_click
    except ImportError:
        pass
    else:
        if isinstance(command, rich_click.RichCommand):
            ctx.console.record = True
            ctx.console.file = open(os.devnull, "w")
```

will work to prevent output to `stdout` while also being invisible to non-rich-click users. Otherwise, will just need to wait for 1.9.
